src/monster.cpp

```diff
@@ -775,7 +775,7 @@ void Monster::doAttacking(uint32_t interval)
 	for (const spellBlock_t& spellBlock : mType->info.attackSpells) {
 		bool inRange = false;
 
-		if (canUseSpell(myPos, targetPos, spellBlock, interval, inRange, resetTicks)) {
+		if (attackedCreature && canUseSpell(myPos, targetPos, spellBlock, interval, inRange, resetTicks)) {
```

Comment:
If `attackedCreature` becomes `nullptr`, there's no point in iterating further:
```suggestion
		if (attackedCreature = nullptr) {
			break;
		}
		if (canUseSpell(myPos, targetPos, spellBlock, interval, inRange, resetTicks)) {
```

Reply:
Why not if (!attackedCreature) ?

Reply:
Just a personal preference, there's no semantic difference.

Reply:
truth, changes already made. Thanks Djarek!